src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.bigquery.stepsdesign;
+
+import io.cdap.e2e.pages.actions.CdfBigQueryPropertiesActions;
+import io.cdap.e2e.pages.actions.CdfStudioActions;
+import io.cdap.e2e.pages.locators.CdfStudioLocators;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.CdfHelper;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.plugin.common.stepsdesign.TestSetupHooks;
+import io.cucumber.java.en.Then;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import stepsdesign.BeforeActions;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * BigQuery related common stepDesigns.
+ */
+public class BigQueryBase implements CdfHelper {
```

Comment:
where is this class used?

Reply:
I think it is containing common step definitions for BQ plugins.

Reply:
Which class inherits from  BigQueryBase? I thought BigQuerySource, BigQuerySink would inherit from BigQueryBase, but they inherit from CdfHelper

Reply:
I don't think cucumber allow inheritance of step definitions.

Reply:
The way to solve this problem is using composition or interfaces @sawantpritam PTAL.

Reply:
Thanks Ankit